internal/provider/resource_gitlab_tag.go

```diff
@@ -0,0 +1,123 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_tag", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "This resource allows you to create and manage GitLab tags.",
+
+		CreateContext: resourceGitlabTagCreate,
+		ReadContext:   resourceGitlabTagRead,
+		DeleteContext: resourceGitlabTagDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name for this tag.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"project": {
+				Description: "The ID or full path of the project which the tag is created against.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"ref": {
+				Description: "The ref which the tag is created from.",
```

Comment:
Let's try an keep using the descriptions from the upstream docs, in this case e.g.:

```suggestion
				Description: "Create tag using commit SHA, another tag name, or branch name",
```

Reply:
Okay